### PR TITLE
Dev/strand guide

### PR DESCRIPTION
**Pull request by anthonykoerber**
@dlasky @shuwen @jcmoore @rpjs 
As your time permits, please have a look

### CODE REVIEW

File: polymer-noconflict.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<script type="text/javascript">
```
Comment (jcmoore) [inline polymer-noconflict.html]: please delete this file

Comment (dlasky) [inline polymer-noconflict.html]: please remove

File: src/strand-guide-canvas/strand-guide-canvas.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * Copyright (c) 2015 MediaMath Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at http://mediamath.github.io/strand/LICENSE.txt
+
+-->
+<link rel="import" href="../../bower_components/polymer/polymer.html">
+
+<dom-module id="strand-guide-canvas">
+	<link rel="import" type="css" href="strand-guide-canvas.css"/>
+	<template>
+		<!-- <content></content> -->
```
Comment (dlasky) [inline src/strand-guide-canvas/strand-guide-canvas.html]: prolly delete this if its not needed (pedantic)

File: src/strand-guide/strand-guide.html [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+<dom-module id="strand-guide">
+	<link rel="import" type="css" href="strand-guide.css"/>
+	<template>
+		<div id="guideContent">
+			<content></content>
+		</div>
+		<template is="dom-if" if="{{useLocalStorage}}">
+			<strand-localstore 
+				id="storage"
+				key="{{name}}"
```
Comment (dlasky) [inline src/strand-guide/strand-guide.html]: we should probably add a namespace to this ie `key="guide-{{name}}"`

File: src/strand-guide-tooltip/strand-guide-tooltip.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+			// Compute next, back, and do labeling
+			this._next = data.length > 1;
+			this._back = data.length > 1 && step > 0;
+
+			if (this._next && step < data.length-1) {
+				this._nextLabel = 'Next';
+			} else if (this._next && step === data.length-1) {
+				this._nextLabel = 'Done';
+			}
```
Comment (jcmoore) [inline src/strand-guide-tooltip/strand-guide-tooltip.js]: This is mostly stylistic, but when `else if` chains I'm left wondering what `else` is not covered -- which requires a bit of cognitive deduction -- and seems to be an unnecessary consideration, so I'd probably do something like:

```
if (this._next) {
  this._nextLabel = (step < data.length - 1) ? 'Next' : 'Done';
}
```

or:

```
if (this._next) {
  if (step < data.length - 1) {
    this._nextLabel = 'Next';
  else {
    this._nextLabel = 'Done';
  }
}
```

**Comment by dlasky** [comment]
LGTM

File: src/strand-guide-tooltip/strand-guide-tooltip.js (hunk 4 above)
Comment (jcmoore) [inline src/strand-guide-tooltip/strand-guide-tooltip.js]: (that, or handle the `step > data.length - 1` case explicitly)

File: src/strand-guide/strand-guide.js [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+		},
+
+		show: function() {
+			if (!this.suppressGuide) {
+				this.hidden = false;
+				this.$.tooltip.open();
+				if (this.showFocus) this.$$('#focus').updateCanvas();
+
+				// strand-modal scroll suppression
+				if(this.noscroll) {
```
Comment (jcmoore) [inline src/strand-guide/strand-guide.js]: careful here, `show()` is a public function, and it looks like repeated calls to `show()` without a `hide()` inbetween could cause undesirable results when `hide()` is called sometime later.  Probably add more conditions beyond just `this.noscroll`

File: src/strand-guide/strand-guide.js [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+				}
+			}
+		},
+
+		hide: function(e) {
+			if (!this.suppressGuide) {
+				this.hidden = true;
+				this.$.tooltip.close();
+
+				if (this.noscroll) {
```
Comment (jcmoore) [inline src/strand-guide/strand-guide.js]: like above, probably do more comprehensive conditioning here (beyond just `this.noscroll`)

**Comment by shuwen** [comment]
This *may* just be a consequence of the test page, but knocking out a circle feels like it also spotlights a bunch of noise:

screen shot 2016-05-02 at 12 19 09 pm

Would it make more sense to highlight only the target element?

File: src/strand-guide/strand-guide.js [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+			},
+			noscroll: {
+				type:Boolean,
+				value:false
+			},
+			closeBtn: {
+				type: Boolean,
+				value: true
+			},
+			_currentStep: {
```
Comment (shuwen) [inline src/strand-guide/strand-guide.js]: Could this use `StrandTraits.Selectable` instead?

**Comment by anthonykoerber** [comment]
CR feedback has been addressed @jcmoore @dlasky @shuwen 

Re: Circle "spotlight" the best (and least unattractive) option IMHO was to switch to an ellipse, which eliminated some of the peripheral 'noise' around the target.

File: src/shared/js/rectangle.html [hunk 8]
```diff
@@ -125,6 +125,29 @@
 			(r.VCENTER & mask) && (this.vcenter = other.vcenter);
 			
 			return this;
+		},
+
+		roundedRectCanvas : function(ctx, r, o) {
```
Comment (dlasky) [inline src/shared/js/rectangle.html]: do we need this for ellipse?

**Comment by dlasky** [comment]
👍

File: src/shared/js/rectangle.html (hunk 8 above)
Comment (anthonykoerber) [inline src/shared/js/rectangle.html]: Nope, we don't - will remove

**Comment by jcmoore** [comment]
I'll allow it, +1  (though you should check out the CI issues)

**Comment by shuwen** [comment]
👍

**Pull request merged**